tests/check/check.go

```diff
@@ -113,6 +113,29 @@ func CheckConfigVersions(operatorVersion, observedVersion, targetVersion string,
 	}
 }
 
+func KeepCheckingWhile(check func(), while func()) {
+	done := make(chan bool)
+
+	go func() {
+		// Perform some long running operation
+		while()
+
+		// Finally close the validator
+		close(done)
+	}()
+
+	// Keep checking while the goroutine is running
+	for {
+		select {
+		case <-done:
+			return
+		default:
+			check()
+		}
+		time.Sleep(time.Second)
```

Comment:
Instead of this you could take a channel out of a time.Ticker (see https://golang.org/pkg/time/#NewTicker) and put it in the select (instead of sleeping), feels a bit more idiomatic